### ISSUE TITLE
Use USER postgres instead of gosu

### ISSUE THREAD

**Issue by kalitad** [kiasaki/docker-alpine-postgres#7]
Hi, maybe can be easier to use 
```
USER postgres
```
before ENTRYPOINT and you can get rid of *gosu* in your script

**Comment by kiasaki** [comment]
Good point, did you try it out? If not I will.

Always nice to remove unnecessary bits of code :D

**Comment by kalitad** [comment]
Yes, I have to change *Dockerfile* a little (I have small discrepancies) but it works

```
FROM gliderlabs/alpine:3.3
RUN apk add --update postgresql && \
        rm -rf /var/cache/apk/*

ENV TERM xterm

ENV LANG en_US.utf-8
ENV PGDATA /var/lib/postgresql/data

RUN mkdir -p $PGDATA && chown -R postgres $PGDATA

COPY docker-entrypoint.sh /

RUN chown postgres /docker-entrypoint.sh
RUN chmod 755 /docker-entrypoint.sh

USER postgres

ENTRYPOINT ["/docker-entrypoint.sh"]

EXPOSE 5432
CMD ["postgres"]
```
and the docker-entrypoint.sh is also simplified - remove all *gosu* and first command (chmod)

**Comment by kiasaki** [comment]
Tried a few things out and the main reason this works for you is because you are not mounting a volume to the data dir, you are simply creating a folder during build, inside the container and setting the right permissions to it.

In the current use case, we need the `chown` on the `/var/lib/postgresql/data` at run time, meaning we need gosu and to be running the entrypoint as a sudoer.

Still leaving this open as I'd like to find a way to remove gosu...

**Comment by frol** [comment]
@kiasaki It is a bit crazy, but you could write a simple program (scriptable languages won't work), which should just make a `chmod` system call; next, you set suid-bit with `root:root` permissions on the executable, so you can run it from a regular user while it will execute as root. Good luck ;)
